src/software/backend/simulation/physics/physics_ball.cpp

```diff
@@ -20,11 +22,14 @@ PhysicsBall::PhysicsBall(std::shared_ptr<b2World> world, const Ball &ball)
     ball_shape.m_radius = BALL_MAX_RADIUS_METERS;
 
     ball_fixture_def.shape = &ball_shape;
-    // These values do not reflect the ball in real life and may still need some tuning.
-    // For now, they are "ideal" values that give the ball perfectly elastic collision
-    // and no friction
-    // TODO: tune values if necessary (#768)
-    ball_fixture_def.density     = 1.0;
+
+    float ball_area          = M_PI * ball_shape.m_radius * ball_shape.m_radius;
+    ball_fixture_def.density = BALL_MASS_KG / ball_area;
```

Comment:
This is an approximation right? Since the mass will be concentrated toward the centre of the sphere? Could use a comment I think

Reply:
It's not an approximation, and mass will be spread evenly over the fixture shape. Added a comment.

Reply:
that'd how it's being simulated but not actually how it would be irl right? If you took a 2d snapshot of a sphere you'd have more mass/area density in the middle of the circle

Reply:
Not in a 2D slice? The center of mass of the sphere will be at its center due to its uniform shape and density, but I don't understand what you mean about more mass/area density in the middle and why that's relevant.

Reply:
Pretty sure the ball could even be a point mass (with a larger shape for collisions) and act the same because it's center of mass is in the same place.

Reply:
will resolve offline